GiniVision/Classes/ImageMetaInformationManager.swift

```diff
@@ -226,20 +227,22 @@ internal struct ImageMetaInformationManager {
         return filteredInformation as MetaInformation
     }
     
-    fileprivate func merge(_ image: UIImage?, withMetaInformation information: MetaInformation?, andCompression compression: CGFloat) -> Data? {
-        guard let image = image else { return nil }
+    fileprivate func generateImage(withMetaInformation information: MetaInformation?, andCompression compression: CGFloat) -> Data? {
+        guard let image = imageData else { return nil }
         guard let information = information else { return nil }
-        guard let imageData = UIImageJPEGRepresentation(image, compression) else { return nil }
-        guard let source = CGImageSourceCreateWithData(imageData as CFData, nil) else { return nil }
-        let count = CGImageSourceGetCount(source)
-        let mutableData = NSMutableData(data: imageData)
-        guard let type = CGImageSourceGetType(source),
-              let destination = CGImageDestinationCreateWithData(mutableData, type, count, nil) else { return nil }
-        for i in 0...count - 1 {
-            CGImageDestinationAddImageFromSource(destination, source, i, information as CFDictionary)
+        
+        let targetData = NSMutableData()
+        
+        let destination = CGImageDestinationCreateWithData(targetData, kUTTypeJPEG, 1, nil)!
+        let source = CGImageSourceCreateWithData(image as CFData, nil)
+        information.setValue(compression, forKey: kCGImageDestinationLossyCompressionQuality as String)
+        
+        let count = CGImageSourceGetCount(source!)
+        for _ in 0...count - 1 {
```

Comment:
I am pretty sure this iteration is not needed as the destination is created with an image count of 1. We ignore the loop counter and if the source has multiple images we add the first one multiple times to the destination. Don't know what happens in this case as destination was created with an image count of 1...

I think the source would contain in our case only one image* so the loop is unnecessary.

\* **BUT** If we are not sure we should do it as before: get the nr of images from the source, pass the nr to the destination and then add all the images from the source to the destination.